javascripts/discourse/initializers/init-discourse-reply-template-component.js.es6

```diff
@@ -118,47 +122,65 @@ function openComposerWithTemplateAndAction(controller, post, wrap) {
     );
     const match = regex.exec(data.raw || "");
 
+    const formatContext = (context) =>
+      context === "title" ? date.format("L") : localDateFormat(date);
+
     const replacers = [
       {
         regex: /(\$today)/g,
-        fn: () => {
+        fn: (context) => {
           const date = moment().startOf("day");
-          return localDateFormat(date);
+          return formatContext(context, date);
         },
       },
       {
         regex: /(\$tomorrow)/g,
-        fn: () => {
+        fn: (context) => {
           const date = moment().add(1, "day").startOf("day");
-          return localDateFormat(date);
+          return formatContext(context, date);
         },
       },
       {
         regex: /(\$week_start)/g,
-        fn: () => {
+        fn: (context) => {
           const date = moment().startOf("isoWeek");
-          return localDateFormat(date);
+          return formatContext(context, date);
         },
       },
       {
         regex: /(\$week_end)/g,
-        fn: () => {
+        fn: (context) => {
           const date = moment().endOf("isoWeek");
-          return localDateFormat(date);
+          return formatContext(context, date);
         },
       },
       {
         regex: /(\$prev_week_start)/g,
-        fn: () => {
+        fn: (context) => {
           const date = moment().subtract(1, "week").startOf("isoWeek");
-          return localDateFormat(date);
+          return formatContext(context, date);
         },
       },
       {
         regex: /(\$prev_week_end)/g,
-        fn: () => {
+        fn: (context) => {
           const date = moment().subtract(1, "week").endOf("isoWeek");
-          return localDateFormat(date);
+          return formatContext(context, date);
+        },
+      },
+      {
+        regex: /(\$username)/g,
+        fn: () => {
+          return currentUser.username;
+        },
+      },
+      {
+        regex: /(\$name)/g,
+        fn: () => {
+          if (currentUser?.name) {
```

Comment:
I don't think you need `?.` here. currentUser should always be available.

And if I am wrong, then we need `?.` on line 174

Reply:
subtle but we need it :open_mouth:... 

name can be nil on brand new accounts and then this was doing a sub of $name -> null (the text) which looks really odd. 174 though is save... every user has a username

Reply:
Agree, `if` is protecting you from sub to null. 

However, currentUser should be always available, so I proposed a change from:
`if (currentUser?.name)` to `if (currentUser.name)`